dotCMS/src/main/java/com/dotcms/ema/proxy/ProxyTool.java

```diff
@@ -0,0 +1,1214 @@
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse send(String url, Map<String, String> params, String method) {
+        return send(url, params, method, DEFAULT_CONTENT_TYPE, null);
+    }
+
+    /**
+     * Sends a request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param method - The Method (POST/GET/PUT/HEAD/DELETE)
+     * @return A ProxyResponse Object
+     * @param creds - A credentials object for authenticated requests. @ from send
+     */
+    public ProxyResponse send(String url, String params, String method, Credentials creds) {
+        return send(url, params, method, DEFAULT_CONTENT_TYPE, creds);
+    }
+
+    /**
+     * Sends a request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param method - The Method (POST/GET/PUT/HEAD/DELETE)
+     * @return A ProxyResponse Object @ from send
+     */
+    public ProxyResponse send(String url, String params, String method) {
+        return send(url, params, method, DEFAULT_CONTENT_TYPE, null);
+    }
+
+    /**
+     * Sends a get request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse send(String url, Map<String, String> params, Credentials creds) {
+        return send(url, params, METHOD_GET, DEFAULT_CONTENT_TYPE, creds);
+    }
+
+    /**
+     * Sends a get request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse send(String url, Map<String, String> params) {
+        return send(url, params, METHOD_GET, DEFAULT_CONTENT_TYPE, null);
+    }
+
+    /**
+     * Sends a get request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object @ from send
+     */
+    public ProxyResponse send(String url, String params, Credentials creds) {
+        return send(url, params, METHOD_GET, DEFAULT_CONTENT_TYPE, creds);
+    }
+
+    /**
+     * Sends a get request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @return A ProxyResponse Object @ from send
+     */
+    public ProxyResponse send(String url, String params) {
+        return send(url, params, METHOD_GET, DEFAULT_CONTENT_TYPE, null);
+    }
+
+    /**
+     * Sends a get request to a url.
+     *
+     * @param url - The URL
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object @ from send
+     */
+    public ProxyResponse send(String url, Credentials creds) {
+        String params = "";
+        if (url.contains("?")) {
+            int idx = url.indexOf("?");
```

Comment:
Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)